src/lib/scan-imports.ts

```diff
@@ -0,0 +1,217 @@
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+import ora from 'ora';
+import glob from 'glob';
+import {parse} from '@babel/parser';
+import traverse, {NodePath} from '@babel/traverse';
+import {
+  CallExpression,
+  ImportDeclaration,
+  isImportDeclaration,
+  isCallExpression,
+} from '@babel/types';
+
+interface DependencyTree {
+  [key: string]: string[];
+}
+
+const cwd = process.cwd();
+
+/**
+ * Renames `web_modules/vue.js` to `vue` based on --dest
+ */
+function getNpmName(modulePath: string, dest: string) {
+  const noExtension = modulePath.replace(/\.[^.]+$/, '');
+
+  // this RegEx allows for web_modules, web_modules/, ./folder/web_modules, etc. as valid --dest options
+  const [lastSegment] = dest.match(/\/?([^/]+)\/?$/);
+  const [_, moduleName] = noExtension.split(lastSegment);
+
+  if (!moduleName) {
+    return noExtension; // module not in --dest
+  }
+
+  const parts = moduleName.split('/').filter(part => part); // split into folders
+  const isScoped = parts[0].startsWith('@');
+  return isScoped ? `${parts[0]}/${parts[1]}` : parts[0];
+}
```

Comment:
So from our discussion, I tried to pass modules like `/web_modules/vue.js` to `install()` and it couldn’t resolve it. This is the least amount of code I could write that passes names to `install()` and have it work out.

I thought of possibly adding a flag like you suggested, but to me it just felt like a really bad user experience (it’s just so easy to get this wrong, and it would require a lot of docs to communicate this setting to users). I think we really can figure this out from `--dest` without getting too complicated, or at least I’d like to try.

The logic here is:
1. Take the last folder segment of `--dest` (assuming it does have a `/`, otherwise, take everything)
2. From that `--dest` folder, assume what follows is an npm package name (if it’s not, then why would it be in the same folder as `--dest`?)

Not too bad to me.

Reply:
As long as we can support projects that use bare-module imports as well, then I think this is fine for the first version. That shouldn't require any config, and I'm assuming would be as simple as "if it's a bare module specifier, it's a dep. Otherwise, do the dest check."

Reply:
This is also unresolved, but cleaning this up and asking in another comment related to new code.